js/viz/graphic_objects.js

```diff
@@ -0,0 +1,19 @@
+import { getNextDefsSvgId } from './core/utils';
+
+const graphicObjects = {};
+
+export const registerPattern = (options) => {
+    const id = getNextDefsSvgId();
+    graphicObjects[id] = { element: 'pattern', ...options };
```

Comment:
It's strange for me to see `element: 'pattern' | 'gradient'`, can we rename it to `type`, for example? 
(same for `registerGradient`)
```suggestion
    graphicObjects[id] = { type: 'pattern', ...options };
```